### PR TITLE
Ensure WithArgs can be used with executable resources

### PR DESCRIPTION
**Pull request by DamianEdwards**
This is follow up to https://github.com/dotnet/aspire/pull/1308 which incorrectly constrained `WithArgs` to `ContainerResource` resources only.

This needs to be backported to preview.2 also.

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/1363)

### CODE REVIEW

File: src/Aspire.Hosting/Publishing/ManifestPublisher.cs [hunk 1]
```diff
@@ -132,6 +132,10 @@ private static void WriteExecutable(ExecutableResource executable, ManifestPubli
         context.Writer.WriteString("workingDirectory", relativePathToProjectFile);
 
         context.Writer.WriteString("command", executable.Command);
+
+        // TODO: This should be calling into any added ExecutableArgsCallbackAnnotation instances on the resource to allow them to
```
Comment (davidfowl) [inline src/Aspire.Hosting/Publishing/ManifestPublisher.cs:136]: Make this change?

Comment (DamianEdwards) [reply src/Aspire.Hosting/Publishing/ManifestPublisher.cs:136]: Already done in #1366

**Comment by DamianEdwards** [comment]
Closing this out after talking to @davidfowl. We don't need to expose this publicly for executables for now.

**Pull request closed without merging**